src/types.rs

```diff
@@ -30,36 +41,301 @@ impl PartialEq for Diagnostic {
   }
 }
 
-impl Diagnostic {
-  /// Message text of the diagnostic.
-  pub fn message(&self) -> Cow<str> {
-    self.kind.msg()
-  }
-
+impl ParseDiagnostic {
   /// 1-indexed display position the diagnostic occurred at.
   pub fn display_position(&self) -> LineAndColumnDisplay {
     self.source.line_and_column_display(self.range.start)
   }
 }
 
-impl Diagnostic {
+impl Diagnostic for ParseDiagnostic {
```

Comment:
Not going to use this in the CLI as part of the larger refactor I'm doing (for simplicitly and not to break things), but we have it for the future.